### ISSUE TITLE
Supabase Docker self-hosting throws 500 errors

### ISSUE THREAD

**Issue by xMdb** [supabase/supabase#5066]
# Bug report

## Describe the bug

The web interface loads but other areas either load infinity or throw various server errors.

## To Reproduce

Follow the steps at Docker https://supabase.com/docs/guides/hosting/docker and head to localhost:3000 as instructed. The web interface (studio) loads but navigating to areas such as tables, API, and database, will show 500 errors.

## Expected behavior

Able to navigate to areas such as tables, API, and database, without seeing 500 errors.

## Screenshots

Below are screenshots of errors I get when running `docker-compose up`.

![image](https://user-images.githubusercontent.com/46137770/150270706-4661c793-a8a1-4523-bb0c-137c05f5b297.png)

![image](https://user-images.githubusercontent.com/46137770/150270770-25cd7543-eedc-450e-91d1-7ad0199bac46.png)

After these errors the images are constantly rebooted due to errors (mostly about connecting to the DB).

On the web panel:

![image](https://user-images.githubusercontent.com/46137770/150271322-c2b4c99c-8820-4775-b02f-2285f973163d.png)

## System information

Ubuntu 20.04 on Raspberry Pi, using docker-compose

**Comment by franklinharvey** [comment]
seeing this both with Docker and the supabase CLI on Mac

**Comment by franklinharvey** [comment]
@xMdb try changing
```env
KONG_URL=http://kong:8000
```
to
```env
KONG_URL=http://localhost:8000
```

in your `docker/env` file

**Comment by xMdb** [comment]
> @xMdb try changing
> 
> ```shell
> KONG_URL=http://kong:8000
> ```
> 
> to
> 
> ```shell
> KONG_URL=http://localhost:8000
> ```
> 
> in your `docker/env` file

Same issue.

**Comment by franklinharvey** [comment]
very strange. Are you sure you're rebuilding? With new a new environment you might need something like `docker compose up --build` or `docker compose up --force-recreate --no-deps` **that might wipe your volume though**

**Comment by FatumaA** [comment]
@xMdb I'm not sure if it's the same issue but I had a similar issue where I could load up the studio using the cli but couldn't interact with it. And could load the studio through the local dev instructions in the readme but had the tables not loading with a similar error

What I did was access the studio through the local dev set up and have the supabase cli running at the same time. Then changed the URL STUDIO_PG_META_URL in the env to the API port that the cli was pointing to. which was localhost:54321 

It's probably a hack but might help

**Comment by rotemrevivo91** [comment]
1. If you have problems loading the api page: 
Pay attention that if you have changed the default ANON_KEY and SERVICE_ROLE_KEY they are also necessary in `docker/volumes/api/kong.yml`.
Further, if you used the ANON_KEY and SERVICE_ROLE_KEY .env variables in kong.yml this will not work and will cause 500 errors when accessing the api page. It doesn't work because the kong.yml config is mounted before the the .env variables are injected to the file. You could use a bash script that does that before `docker-compose build` is used.
I'm using doppler for my implementation:
```sh
#!/bin/sh

# Prevent configure command being leaked in bash history
export HISTIGNORE='doppler*'

# Scope to location of application directory
echo '<doppler_key>' | doppler configure set token --scope ~/Projects/supabase/docker

ANON_KEY=$(doppler run -- printenv | grep ANON_KEY)
ANON_KEY=${ANON_KEY//'ANON_KEY='}
SERVICE_ROLE_KEY=$(doppler run -- printenv | grep SERVICE_ROLE_KEY)
SERVICE_ROLE_KEY=${SERVICE_ROLE_KEY//'SERVICE_ROLE_KEY='}

# attach anon_key and service_role_key to kong.yml before mounting it
doppler run --command="sed -e 's/REPLACE_ANON_KEY/$ANON_KEY/g' -e 's/REPLACE_SERVICE_KEY/$SERVICE_ROLE_KEY/g' volumes/api/kong.yml" > volumes/api/kongi.yml  

# Supply secrets to your application
doppler run -- docker-compose up -d

# remove kongi.yml
rm volumes/api/kongi.yml
```
Then, in the docker-compose.yml file under the kong service change this:
```yml
    volumes:
      - ./volumes/api/kong.yml:/var/lib/kong/kong.yml
``` 
To:
```yml
    volumes:
      - ./volumes/api/kongi.yml:/var/lib/kong/kong.yml
```

2. If you have problems loading other pages (database, table editor and running sql scripts) that means that the addresses pointing to kong in wrong. In development the default values of: http://kong:8000, http://meta:8080 should remain. In a production environment you'll have to give kong an actual url.

**Comment by mellkior** [comment]
> @xMdb try changing
> 
> ```shell
> KONG_URL=http://kong:8000
> ```
> 
> to
> 
> ```shell
> KONG_URL=http://localhost:8000
> ```
> 
> in your `docker/env` file

This made the `api` area of studio work for me while self-hosting on windows using the guide [here](https://supabase.com/docs/guides/hosting/docker).

**Comment by xMdb** [comment]
> very strange. Are you sure you're rebuilding? With new a new environment you might need something like `docker compose up --build` or `docker compose up --force-recreate --no-deps` **that might wipe your volume though**

(sorry for the late reply, been busy)

@franklinharvey I did `docker-compose` with `--build` and it worked, but also after wiping my images/volumes and omitting the `--build` it worked. Not sure what happened, could have been those old images, my Docker version, or someone fixing it since the last time I tried (which was when I opened this issue)

> 1. If you have problems loading the api page:
>        Pay attention that if you have changed the default ANON_KEY and SERVICE_ROLE_KEY they are also necessary in `docker/volumes/api/kong.yml`.

@rotemrevivo91 this also could have been my issue, as this time I re-cloned the repo and made sure to edit the kong.yml file.

I'm going to close this issue, as either wiping the images/volumes, using `docker-compose up` with `--build`, updating Docker, or changing the values in `kong.yml` was obviously the issue.
